### ISSUE TITLE
Inquiry about Training Configurations (Batch Size, LR, Commit Hash) for Replicating SmolVLA on LIBERO Benchmark

### ISSUE THREAD

**Issue by Yoonkyo** [huggingface/lerobot#3287]
Hi LeRobot team,

First of all, thank you for the amazing work on SmolVLA and the LeRobot library!

I am currently trying to reproduce the SmolVLA results on the LIBERO benchmark, but I am observing a significant gap between my evaluation results and the performance reported in the paper. 

According to the paper, the 0.45B model achieves success rates of approximately 90%, 96%, 92%, 71% (Spatial, Object, Goal, Long), averaging 87.3%. However, my current fine-tuning setup yields lower results:

**My Evaluation Results:**
* **libero_10:** 40.6%
* **libero_goal:** 76.0%
* **libero_object:** 71.0%
* **libero_spatial:** 68.0%

**My Training Setup:**
* **Base model:** Fine-tuned from `smolvla_base` (ensured `load_vlm_weights=True`)
* **Policy type:** SmolVLA
* **Batch size:** 32 
* **Steps:** 80,000

I suspect the discrepancy might be primarily due to the effective batch size. Since I am training with a batch size of 32, I was wondering if the paper's results were achieved using a much larger effective batch size via multi-GPU training. 

Could you please share the exact training configurations used for the LIBERO evaluations in the paper? Specifically, I would greatly appreciate details on:
1. **Effective / Global Batch Size** (and the number of GPUs used)
2. **Learning Rate** (and LR scheduler if applicable)
3. **The specific commit hash** used for the paper's experiments (I would like to restore that exact version to rule out any codebase changes).
4. Any other specific hyperparameter tweaks used for the LIBERO benchmark.

Thank you in advance for your time and guidance!

**Comment by Yoonkyo** [comment]
I was able to resolve most of the reproduction gap, so I’m closing this issue.

The main problem was the learning-rate schedule. In this version of LeRobot, `scheduler_decay_steps` defaults to `30000`, independently of the total number of training steps. As a result, in a 100k-step run, the learning rate reached the minimum value at 30k and remained near `2.5e-6` for the remaining 70% of training.

After explicitly setting:

```bash
--policy.scheduler_decay_steps=100000
```

and matching the paper architecture more carefully, the results improved substantially.

The other important details were:

* Build a fresh SmolVLA policy with `--policy.type=smolvla`
* Load only the pretrained SmolVLM2 backbone with `--policy.load_vlm_weights=true`
* Use `expert_width_multiplier=0.75`
* Use the first 16 VLM layers
* Use a 100-step warmup
* Train a single multi-task model over all 40 LIBERO tasks
* Use a global batch size of 64

My final training command was:

```bash
accelerate launch \
  --num_processes=2 \
  --num_machines=1 \
  --mixed_precision=bf16 \
  --main_process_port=29511 \
  --module lerobot.scripts.lerobot_train \
  --policy.type=smolvla \
  --policy.load_vlm_weights=true \
  --policy.expert_width_multiplier=0.75 \
  --policy.num_vlm_layers=16 \
  --policy.freeze_vision_encoder=true \
  --policy.train_expert_only=true \
  --policy.chunk_size=50 \
  --policy.n_action_steps=10 \
  --policy.num_steps=10 \
  --policy.optimizer_lr=1e-4 \
  --policy.scheduler_warmup_steps=100 \
  --policy.scheduler_decay_steps=100000 \
  --policy.scheduler_decay_lr=2.5e-6 \
  --policy.push_to_hub=false \
  --dataset.repo_id=HuggingFaceVLA/libero \
  --dataset.revision=v3.0 \
  --batch_size=32 \
  --steps=100000 \
  --num_workers=6 \
  --save_freq=10000 \
  --env_eval_freq=0 \
  --wandb.enable=true \
  --wandb.project=smolvla_libero_repro \
  --job_name=smolvla_libero_paperrecipe_gb64_100k_decay100k \
  --output_dir=<OUTPUT_DIR> \
  --seed=1000
```

With this corrected setup, my final 100k checkpoint reached:

```text
Spatial: 78.7%
Object:  92.7%
Goal:    89.0%
Long:    60.3%
Overall: 80.2%
```

The best intermediate checkpoint, at 60k steps, reached:

```text
Spatial: 78.0%
Object:  94.3%
Goal:    89.0%
Long:    66.3%
Overall: 81.9%
```

These results still do not exactly reproduce the paper’s reported numbers, especially on LIBERO-Spatial and LIBERO-Long, but they are substantially better than my original run, where LIBERO-Long was around 43–45%.

The `scheduler_decay_steps=100000` override was the largest factor in recovering performance. Using `--policy.path=smolvla_base` should also be avoided for this reproduction, since its saved configuration may override the paper architecture with different expert width and VLM depth values. Building a fresh config with `--policy.type=smolvla` was important. 

Thank you again for the library and for making the models available!
